torch/_inductor/mkldnn_lowerings.py

```diff
@@ -889,6 +896,25 @@ def qlinear_binary(
                 )
             else:
                 x_scale.realize()
+
+            x_scale_size = x_scale.get_size()
+            if len(x_scale_size) > 1:
+                if x_scale_size[0] == 1 and x_scale_size[1] == 1:
+                    # Corner-case discovered with LLaMA series.
+                    # If x_scale is shaped [1, 1], make it a 0D tensor.
+                    # Otherwise, epilogue creator will run into indexing issues.
+                    x_scale = view(x_scale, [])
+            assert len(x_scale.get_size()) in [0, 1], "x_scale must be 0D or 1D"
+
+            if x_zp is None:
+                x_zp = V.graph.add_tensor_constant(
+                    torch.tensor(0, dtype=torch.int32), name="x_zp"
+                )
+
+            if w_zp is None:
+                w_zp = V.graph.add_tensor_constant(
+                    torch.tensor(0, dtype=torch.int32), name="w_zp"
+                )
```

Comment:
This particular change is now necessary because of [an unnecessary change](https://github.com/pytorch/pytorch/pull/143187#discussion_r1921052467) introduced in https://github.com/pytorch/pytorch/pull/143187/commits/09bdb53dc849735714d767fd81d59b3c402da38f